spec/lib/adviser_list_csv_spec.rb

```diff
@@ -1,7 +1,34 @@
 RSpec.describe AdviserListCsv do
   describe '#to_csv' do
+    let(:qualifications) { 3.times.map { FactoryGirl.create(:qualification) } }
+    let(:accreditations) { 3.times.map { FactoryGirl.create(:accreditation) } }
+    let(:advisers) { [adviser1, adviser2] }
+    let(:adviser1) { FactoryGirl.create(:adviser, qualifications: [qualifications.first], accreditations: [accreditations.first]) }
+    let(:adviser2) { FactoryGirl.create(:adviser, qualifications: [qualifications.last], accreditations: [accreditations.last]) }
+    let(:subject)  { described_class.new(advisers) }
+
     it 'responds to csv conversion' do
-      expect(described_class.new []).to respond_to(:to_csv)
+      expect(subject).to respond_to(:to_csv)
+    end
+
+    it 'produces correct data' do
```

Comment:
My feeling is that this spec is become a little bigger. Can we break into smaller specs? One for the header and one for the data? OR can we rename this to something related to: 'returns qualifications from each advisor'?

Reply:
I've made a couple of changes to the spec @tomas-stefano, please could you have another look and see if this is any clearer?

Reply:
Sorry to be picky but can you rename the specs to something like "returns csv headings" instead of using "has correct headings"? And the same for advisers data specs. I am not sure about others, but "has correct ..." on specs sounds redundant to me.

Reply:
No worries, can do :)